contracts/tokens/SecurityToken.sol

```diff
@@ -515,29 +507,23 @@ contract SecurityToken is StandardToken, DetailedERC20, ReentrancyGuard, Registr
     }
 
     /**
-     * @notice End token minting period permanently for Issuer
-     */
-    function finishMintingIssuer() external onlyOwner {
-        finishedIssuerMinting = true;
-        emit LogFinishMintingIssuer(now);
-    }
-
-    /**
-     * @notice End token minting period permanently for STOs
+     * @notice Permanently freeze minting of this security token.
+     * @dev It MUST NOT be possible to increase `totalSuppy` after this function is called.
      */
-    function finishMintingSTO() external onlyOwner {
-        finishedSTOMinting = true;
-        emit LogFinishMintingSTO(now);
+    function freezeMinting() external mintingAllowed() onlyOwner {
+        require(IModuleRegistry(moduleRegistry).freezeMintingAllowed());
+        mintingFrozen = true;
+        emit LogFreezeMinting(now);
     }
 
     /**
      * @notice mints new tokens and assigns them to the target _investor.
-     * @dev Can only be called by the STO attached to the token (Or by the ST owner if there's no STO attached yet)
-     * @param _investor Address to whom the minted tokens will be dilivered
-     * @param _amount Number of tokens get minted
+     * @dev Can only be called by the issuer or STO attached to the token
+     * @param _investor Address where the minted tokens will be delivered
+     * @param _amount Number of tokens be minted
      * @return success
      */
-    function mint(address _investor, uint256 _amount) public onlyModule(STO_KEY, true) checkGranularity(_amount) isMintingAllowed() returns (bool success) {
+    function mint(address _investor, uint256 _amount) public onlyModule(STO_KEY, true) checkGranularity(_amount) mintingAllowed() returns (bool success) {
```

Comment:
There's a big change missing here. Mint has the onlyModule modifier, which still prevents minting from happening if there's an STO attached. onlyModule logic has to be reviewed. 
Tests should have caught this... Please review why the tests didn't and fix them.

Reply:
I believe this was a bug we decided not to fix. I definitely think the logic of onlyModule needs to be simplified. I've added a task to that effect

Reply:
Current functionality is that minting is not possible while there's an STO attached. As per the task at hand we need to change it so onlyModule doesn't prevent minting when there's an STO.